src/tcp_wrap.cc

```diff
@@ -181,6 +183,16 @@ TCPWrap::TCPWrap(Environment* env, Local<Object> object, ProviderType provider)
                    // Suggestion: uv_tcp_init() returns void.
 }
 
+void TCPWrap::Reinitialize(const FunctionCallbackInfo<Value>& args) {
+  TCPWrap* wrap;
+
+  ASSIGN_OR_RETURN_UNWRAP(
+      &wrap, args.Holder(), args.GetReturnValue().Set(UV_EBADF));
+
+  Environment* env = wrap->env();
+  int r = uv_tcp_init(env->event_loop(), &wrap->handle_);
+  CHECK_EQ(r, 0);
```

Comment:
nit: since this bit of code is also used in the TCPWrap constructor - https://github.com/nodejs/node/blob/ecd385e891d968d6987dc3629ea2a4efdb2639cd/src/tcp_wrap.cc#L177-L182, maybe we could create a method with this code and call that in both places?

Reply:
That's a really old comment, by the way; you might as well remove it. `uv_tcp_init()` is effectively infallible.

(`uv_tcp_init_ex()` is a different story but node doesn't use that.)